Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,423 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	public:
+		/**
+		 * WireGuard message types
+		 */
+		enum class WireGuardMessageType
+		{
+			/** Handshake Initiation message */
+			HandshakeInitiation = 1,
+			/** Handshake Response message */
+			HandshakeResponse = 2,
+			/** Cookie Reply message */
+			CookieReply = 3,
+			/** Transport Data message */
+			TransportData = 4
+		};
+
+#pragma pack(push, 1)
+		/**
+		 * @struct wg_common_header
+		 * Represents the common header for all WireGuard message types
+		 */
+		struct wg_common_header
+		{
+			/** Message type field */
+			uint8_t messageType;
+			/** Reserved field (3 bytes) */
+			uint8_t reserved[3];
+		};
+#pragma pack(pop)
+
+		WireGuardLayer()
+		{}
+
+		/**
+		 * Constructs a WireGuardLayer object.
+		 *
+		 * @param data Pointer to the raw data representing the WireGuard layer
+		 * @param dataLen Length of the data
+		 * @param prevLayer Pointer to the previous layer in the packet (if any)
+		 * @param packet Pointer to the packet this layer belongs to
+		 */
+		WireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		    : Layer(data, dataLen, prevLayer, packet, Wireguard)
+		{}
+
+		wg_common_header* getBasicHeader() const
+		{
+			return (wg_common_header*)m_Data;
+		}
+
+		/**
+		 * Checks if the given port numbers are WireGuard ports.
+		 *
+		 * @param portSrc The source port number to check
+		 * @param portDst The destination port number to check
+		 * @return True if either port matches the WireGuard port (51820), false otherwise
+		 */
+		static bool isWireguardPorts(uint16_t portSrc, uint16_t portDst)
+		{
+			return (portSrc == 51820 || portDst == 51820);
+		}
+
+		/**
+		 * Checks if the given data represents a WireGuard message.
+		 *
+		 * @param data Pointer to the raw data
+		 * @param dataLen Length of the data
+		 * @return True if the data starts with a valid WireGuard message type, false otherwise
+		 */
+		static bool isDataValid(const uint8_t* data, size_t dataLen);
+
+		static WireGuardLayer* parseWireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		std::string getMessageTypeAsString() const;
+
+		uint32_t getMessageType()
+		{
+			return getBasicHeader()->messageType;
+		}
+
+		uint8_t* getReserved()
```

Comment:
1. `reserved` is always 3 bytes, so maybe it should return `uint32_t`?
2. This method can be `const`